File: csharp/1556.thousand-separator.cs
/*
 * @lc app=leetcode id=1556 lang=csharp
 *
 * [1556] Thousand Separator
 */

using System.Collections.Generic;
using System.Linq;

// @lc code=start
public class Solution {
    public string ThousandSeparator(int n) {
        if (n == 0) {
            return "0";
        }
        var ret = new List<char>();
        int k = 0;
        while (n > 0) {
            ret.Add((char)((int)'0' + n % 10));
            n /= 10;
            if (++k == 3) {
                k = 0;
                ret.Add('.');
            }
        }
        return new string(ret.Reverse<char>().Skip(k == 0 ? 1 : 0).ToArray());
    }
}
// @lc code=end

File: java/1556.thousand-separator.java
/*
 * @lc app=leetcode id=1556 lang=java
 *
 * [1556] Thousand Separator
 */

// @lc code=start
class Solution {
    public String thousandSeparator(int n) {
        if (n == 0) {
            return "0";
        }
        StringBuilder sb = new StringBuilder();
        while (n > 0) {
            if ((sb.length() - 3) % 4 == 0) {
                sb.insert(0, '.');
            }
            sb.insert(0, n % 10);
            n /= 10;
        }
        return sb.toString();
    }
}
// @lc code=end

